patches/server/0731-Workaround-for-Client-Lag-Spikes-MC-162253.patch

```diff
@@ -0,0 +1,233 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MeFisto94 <MeFisto94@users.noreply.github.com>
+Date: Thu, 22 Jul 2021 15:20:17 +0100
+Subject: [PATCH] Workaround for Client Lag Spikes (MC-162253)
+
+When crossing certain chunk boundaries, the client needlessly
+calculates light maps for chunk neighbours. In some specific map
+configurations, these calculations cause a 500ms+ freeze on the Client.
+
+This patch basically serves as a workaround by sending light maps
+to the client, so that it doesn't attempt to calculate them.
+This mitigates the frametime impact to a minimum (but it's still there).
+
+Co-authored-by: basedAF <86579232+basedAF@users.noreply.github.com>
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index f101302892a7f95ffa0e6d63cb69d8a3ddbd67a7..45bee75d2c5f111cc8f6013e0d90c89b6e579454 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -511,4 +511,9 @@ public class PaperConfig {
+         itemValidationBookAuthorLength = getInt("settings.item-validation.book.author", itemValidationBookAuthorLength);
+         itemValidationBookPageLength = getInt("settings.item-validation.book.page", itemValidationBookPageLength);
+     }
++
++    public static boolean workaroundMC162253 = true;
++    private static void workaroundMC162253() {
++        workaroundMC162253 = getBoolean("settings.workaround-MC-162253", workaroundMC162253);
++    }
+ }
+diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
+index 42fd259f4492e539112b5bcb310aaaadab58a443..ea884f6f05a5b066779e88dd524a74b47eee0ebc 100644
+--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
++++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
+@@ -505,7 +505,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
+                     player.needsChunkCenterUpdate = false;
+                     player.connection.send(new ClientboundSetChunkCacheCenterPacket(currPosX, currPosZ));
+                 }
+-                ChunkMap.this.updateChunkTracking(player, new ChunkPos(rangeX, rangeZ), new Packet[2], false, true); // unloaded, loaded
++                ChunkMap.this.updateChunkTracking(player, new ChunkPos(rangeX, rangeZ), new Packet[10], false, true); // unloaded, loaded
+             },
+             (ServerPlayer player, int rangeX, int rangeZ, int currPosX, int currPosZ, int prevPosX, int prevPosZ,
+              com.destroystokyo.paper.util.misc.PooledLinkedHashSets.PooledObjectLinkedOpenHashSet<ServerPlayer> newState) -> {
+@@ -2151,32 +2151,128 @@ Sections go from 0..16. Now whenever a section is not empty, it can potentially
+         return mask;
+     }
+ 
+-    private static int getCeilingLightMask(final LevelChunk chunk) {
```

Comment:
You might as well remove the old method from the patch it's added, since pretty sure it goes unused otherwise.